keps/sig-release/3744-stay-on-supported-go-versions/README.md

```diff
@@ -0,0 +1,418 @@
+it seemed like a good time to capture requirements and a process for updating Kubernetes release branches.
+
+### Goals
+
+* improve security of Kubernetes patch releases by building and releasing using a supported minor version of go
+* preserve stability of Kubernetes patch releases by avoiding regressions or behavior changes due to updating go minor versions
+* avoid end-user action-required items in patch releases due to updating go minor versions
+
+### Non-Goals
+
+* change the cadence of Kubernetes patch releases
+* change the duration of support for a given Kubernetes minor version
+
+## Proposal
+
+Track changes made to the default Kubernetes development branch that were required to adopt a new go minor version (go 1.N).
+This typically includes changes like:
+
+* updates to static analysis tooling to support any go language changes
+  (e.g. [#115129](https://github.com/kubernetes/kubernetes/pull/115129))
+* updates to dependencies needed to work with go 1.N
+  (e.g. [#114766](https://github.com/kubernetes/kubernetes/pull/114766))
+* updates to Kubernetes code to fix issues identified by improved vet or lint checks
+* updates to Kubernetes code to work with both go 1.N and 1.(N‑1)
+  (e.g. [commit c31cc5ec](https://github.com/kubernetes/kubernetes/commit/c31cc5ec46315a02343ec6d6a2ef659e2cc8668e))
+
+Merge those changes to the default Kubernetes development branch *prior* to updating to go 1.N.
+This ensures those changes build and pass tests with both go 1.N and 1.(N‑1).
+Here is an [example of tracking and pre-merging changes for adopting go 1.20](https://github.com/kubernetes/release/issues/2815#issuecomment-1373891562).
+
+Ensure all of those changes are backported to supported Kubernetes release branches,
+and update those release branches to build/release using go 1.N once all of these
+conditions are satisfied:
+
+1. go 1.N has been released at least 3 months
+   * this gives ~3 months for adoption of go 1.N by the go community
+   * this gives ~3 months for completing the backports and cutting Kubernetes patch releases
+     until go 1.(N+1) is released and go 1.(N‑1) previously used by Kubernetes patch releases
+     goes out of support
+2. go 1.N has been used in a released Kubernetes version for at least 1 month
+   * this ensures all release-blocking and release-informing CI has run on go 1.N
+   * this gives time for release candidates and early adoption by the Kubernetes community
+3. Backported code and dependency changes build and pass unit, integration, and e2e tests with both go 1.N and 1.(N‑1)
+   * this ensures consumers of patch releases of published Kubernetes
+     libraries are not *forced* to update to go 1.N
+4. There are no regressions relative to go 1.(N‑1) known to impact Kubernetes
+5. Behavior changes in go 1.N are mitigated to preserve existing behavior for previous
```

Comment:
It'd be helpful to be able to easily track/clearly communicate in a standard way what Go behavior compatibility requirements are associated with what Kubernetes versions. Any thoughts on how that might be done in practice? Through the changelog and issue labeling? I'd love to be able to look at the changelog (or labels on issues linked to by the changelog) for a given release and explicitly know where go behavioral changes have been mitigated to support a go version bump.

Reply:
at the point where the branch changes to build/release with a new go version, the https://github.com/kubernetes/kubernetes/blob/master/.go-version file in the root would update to the new go version

for tracking backport status of the individual changes/prereqs to support that go version, presumably the tracking issue mentioned above could be used?

Reply:
pushed a commit clarifying the distinct steps